test/plugins/retry.js

```diff
@@ -289,12 +290,11 @@ describe('Retry Plugin', function() {
             assert.equal(resp.statusCode, 200);
           })
           .on('data', function(data) {
-            dataCount++;
-            assert.ok(data.toString('utf8').indexOf('"doc_count":0') > -1);
+            responseString += data;
           })
           .on('end', function() {
+            assert.ok(responseString.toString('utf8').indexOf('"doc_count":0') > -1);
```

Comment:
Do we even need to collect the chunks and assert the string here and then assert the object again later in the `on('finish'...`?
Hmm actually I suppose it does mean that we can distinguish between a failure of "expected `data` didn't arrive" vs a failure of the "plugin didn't work correctly with `pipe`".